### PR TITLE
Split syntax and snippets

### PR DESCRIPTION
**Pull request by gabro**
This PR is a proposal for splitting the syntax and the snippets into different packages.

The reasoning behind this is that I believe the syntax extension should as minimal as possible and only provide the grammar, while snippets should be opt-in, since they're quite annoying for the users if they trigger unexpectedly. 

This is similar to what other languages do on the Marketplace.

Note that I've made the snippets plugin to depend on the syntax one, so that you can't install snippets without the syntax, which would be a cause of confusion.

### CODE REVIEW

File: snippets/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+[VS Code plugin](https://marketplace.visualstudio.com/items?itemName=scala-lang.scala-snippets)
+
+## Features
+
+Extension providing Scala code snippets.
+
+## Requirements
+
+No requirements.
```
[the remark this answers is not in the document]
Comment (gabro) [reply snippets/README.md]: it should, that's an oversight.

If there's still interest in merging this PR I can update it and rebase it.

[the remark this answers is not in the document]
Comment (Duhemm) [reply snippets/README.md]: Yes please! Thanks

[the remark this answers is not in the document]
Comment (gabro) [reply snippets/README.md]: @Duhemm done!

**Comment by gabro** [comment]
Thank you for merging, @Duhemm! Will you take care of publishing the new plugins on the Marketplace?

**Comment by nicolasstucki** [comment]
I took care of publishing

**Comment by gabro** [comment]
Thank you, @nicolasstucki!

**Pull request merged**